test/e2e/dynamic_provisioning_test.go

```diff
@@ -141,7 +151,50 @@ func (t *dynamicProvisioningTestSuite) defineTests(isMultiZone bool, schedulerNa
 				"logicalSectorSize": "512",
 			}
 		}
-		test.Run(cs, ns, schedulerName)
+		test.Run(cs, ns, azDiskClient, schedulerName, isUsingCSIDriverV2)
+	})
+
+	ginkgo.It(fmt.Sprintf("should create a volume on demand with mount options with maxShares > 1 [kubernetes.io/azure-disk] [disk.csi.azure.com] [Windows] [%s]", schedulerName), func() {
+		pvcSize := "256Gi"
+		if isMultiZone {
+			pvcSize = "1000Gi"
+		}
+		pods := []testsuites.PodDetails{
+			{
+				Cmd: convertToPowershellorCmdCommandIfNecessary("echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data"),
+				Volumes: t.normalizeVolumes([]testsuites.VolumeDetails{
+					{
+						ClaimSize: pvcSize,
+						MountOptions: []string{
+							"barrier=1",
+							"acl",
+						},
+						VolumeMount: testsuites.VolumeMountDetails{
+							NameGenerate:      "test-volume-",
+							MountPathGenerate: "/mnt/test-",
+						},
+					},
+				}, isMultiZone),
+				IsWindows: isWindowsCluster,
+			},
+		}
+		test := testsuites.DynamicallyProvisionedCmdVolumeTest{
+			CSIDriver:              testDriver,
+			Pods:                   pods,
+			StorageClassParameters: map[string]string{"skuName": "Standard_LRS"},
+		}
+
+		if isMultiZone && !isUsingInTreeVolumePlugin {
+			test.StorageClassParameters = map[string]string{
+				"skuName":           "UltraSSD_LRS",
+				"cachingmode":       "None",
+				"diskIopsReadWrite": "2000",
+				"diskMbpsReadWrite": "320",
+				"logicalSectorSize": "512",
+				"maxShares":         "2",
+			}
+		}
+		test.Run(cs, ns, azDiskClient, schedulerName, isUsingCSIDriverV2)
```

Comment:
Rather than passing a Boolean flag to each method, consider adding an `isUsingCSIDriverV2() bool` function to `suite_test.go` and using that to customize test cases for V2 testing.

Reply:
I also took this approach at the starting stage but realized that this would cause an import cycle as `e2e` package, which is where `isUsingCSIDriverV2` variable is set, has dependency on `testsuite` package, which is the package all the tester files (e.g. `DynamicallyProvisionedCmdVolumeTest` belongs to.

Reply:
Ah, yes, I see now. Yeah, passing as an argument is what needs to happen here.